mysql/const.go

```diff
@@ -425,14 +429,20 @@ const (
 	ModePadCharToFullLength
 )
 
-// GetSQLMode gets the sql mode for string literal.
-func GetSQLMode(str string) SQLMode {
-	str = strings.ToUpper(str)
-	mode, ok := Str2SQLMode[str]
-	if !ok {
-		return ModeNone
+// GetSQLMode gets the sql mode for string literal. SQL_mode is a list of different modes separated by commas.
+func GetSQLMode(s string) (SQLMode, error) {
+	strs := strings.Split(s, ",")
+	strs[len(strs)-1] = strings.TrimRight(strs[len(strs)-1], " ")
```

Comment:
Just trim `s` is simpler.

Reply:
I noticed that the input string `s` is already formatted, we don't need to do it again.

Reply:
OK.  I'll move `FormatSQLModeStr` beside `GetSQLMode` because they are closely related. Besides, we'd better add a comment for `GetSQLMode` to  de denote that the input of `GetSQLMode` must be formatted by `FormatSQLModeStr`